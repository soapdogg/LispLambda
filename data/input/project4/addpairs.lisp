(DEFUN ADDPAIRS (X Y Z)
    (COND
        ((NULL Y)
            Z
        )
        ((ATOM X)
            Z
        )
        ((NULL(ATOM(CAR X)))
                    Z
        )
        ((ATOM Y)
            Z
        )
        ((NULL X)
            Z
        )
        (T
            (CONS
                (CONS
                    (CAR X)
                    (CAR Y)
                )
                (ADDPAIRS (CDR X) (CDR Y) Z)
            )
        )
    )
)

(DEFUN GETVAL (X Z)
    (COND
        ((EQ X (CAR (CAR Z)))
            (CDR (CAR Z))
        )
        (T
            (GETVAL X (CDR Z))
        )
    )
)

(ADDPAIRS (CONS 3 NIL) (CONS 34 NIL) ())
(ADDPAIRS (CONS 45 NIL) (CONS 56 NIL) (QUOTE ()))
(ADDPAIRS
    (CONS 45 (CONS 89 (CONS 34 NIL)))
    (CONS (QUOTE (23 44 45)) (CONS NIL (CONS (QUOTE (56 34)) NIL)))
    (CONS (CONS 34 34) (CONS (CONS 67 12) NIL)))