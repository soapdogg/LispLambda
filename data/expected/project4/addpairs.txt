((3 . 34))
((45 . 56))
((45 23 44 45) (89) (34 56 34) (34 . 34) (67 . 12))